extension/src/git.ts

```diff
@@ -51,6 +51,17 @@ export const getAllUntracked = async (
   return new Set([...files, ...dirs])
 }
 
+export const getHasChanges = async (
+  repositoryRoot: string
+): Promise<boolean> => {
+  const output = await executeProcess({
+    args: ['diff', '--name-status', '-z'],
```

Comment:
[F] The first bug was here. `git status` touches `.git/index` which fired the watcher which ran `git status` which...